runner/pkg/watchman/cache.go

```diff
@@ -109,7 +120,13 @@ func (c *watchmanCache) read() {
 	c.old = v.Entries
 	c.lastClockSpec = cs.ClockSpec
 
-	BazelLog.Infof("Watchman cache: %d entries at clock spec %s", len(c.old), c.lastClockSpec)
+	// Persist the fact that all persisted paths are not symlinks.
```

Comment:
This ensures the new `LStat` call does not add extra overhead for cached/incremental runs. Since we only persist files watchman is aware of (real files, not symlinks) we know everything cached and not invalidated is a real non-symlink file and we can avoid the `LStat` later on.

Without this the `LStat` was a performance regression on large repos.